File: Array & String/Best Time to Buy and Sell Stock/BuySellStock1.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int profit = 0;
        int minPrice = int.MaxValue;
        for (int i = 0; i < prices.Length; i++) {
            minPrice = Math.Min(minPrice, prices[i]);
            profit = Math.Max(profit, prices[i] - minPrice);
        }
        return profit;
    }
}

File: Array & String/Best Time to Buy and Sell Stock/BuySellStock1.java
class Solution {

  public int maxProfit(int[] prices) {
    int minVal = Integer.MAX_VALUE;
    int maxProfit = 0;

    for (int i = 0; i < prices.length; i++) {
      if (prices[i] < minVal) {
        minVal = prices[i];
      } else if (prices[i] - minVal > maxProfit) {
        maxProfit = prices[i] - minVal;
      }
    }
    return maxProfit;
  }
}
